app/src/main/java/de/grobox/transportr/utils/TransportrUtils.java

```diff
@@ -649,6 +649,22 @@ static public int getDragDistance(Context context) {
 		return (int) (30 * metrics.density);
 	}
 
+	public static Intent getShortcutIntent(Context context, long uid, WrapLocation from, @Nullable WrapLocation via, WrapLocation to) {
+		Intent shortcutIntent = new Intent(context, DirectionsActivity.class);
+		shortcutIntent.setAction(ACTION_MAIN);
+		shortcutIntent.addFlags(FLAG_ACTIVITY_NEW_TASK);
+		shortcutIntent.addFlags(FLAG_ACTIVITY_CLEAR_TOP);
+
+		//FIXME Doesn't work if they are not null
+		shortcutIntent.putExtra(FAV_TRIP_UID, uid);
+		shortcutIntent.putExtra(FROM, from);
+		shortcutIntent.putExtra(VIA, via);
+		shortcutIntent.putExtra(TO, to);
+		shortcutIntent.putExtra(SEARCH, false);
```

Comment:
My guess is that you can't put complex objects that need to be serialized into shortcut intents.

Reply:
If that's true, you could work with `StoredSearch` `uid`s, but you need to handle the case when the `uid` doesn't exist anymore.